classes/Tools.php

```diff
@@ -968,28 +968,19 @@ public static function convertPriceFull($amount, Currency $currency_from = null,
      */
     public static function dateFormat($params, &$smarty)
     {
-        return Tools::displayDate($params['date'], null, (isset($params['full']) ? $params['full'] : false));
+        return Tools::displayDate($params['date'], (isset($params['full']) ? $params['full'] : false));
     }
 
     /**
      * Display date regarding to language preferences.
      *
      * @param string $date Date to display format UNIX
-     * @param int $id_lang Language id DEPRECATED
      * @param bool $full With time or not (optional)
-     * @param string $separator DEPRECATED
      *
      * @return string Date
      */
-    public static function displayDate($date, $id_lang = null, $full = false, $separator = null)
+    public static function displayDate($date, $full = false)
```

Comment:
Isn't this going to make it very hard for modules to be compatible with 1.7 and 8?

The only way to work around this is to have a `if()` to know whether to put your 3rd argument in 2nd or 3rd position

Reply:
It's the same with all deprecated methods that we remove and used many parameters.
Since the method has been marked as deprecated for years (9 and 11years exactly), it's time to clean up this.